core/src/main/antlr4/com/databricks/labs/remorph/parsers/snowflake/SnowflakeParser.g4

```diff
@@ -3681,7 +3681,7 @@ expr_list_sorted
 expr
     : object_name DOT NEXTVAL
     | expr LSB expr RSB //array access
-    | expr COLON expr   //json access
+    | expr COLON json_path   //json access
     | expr DOT (VALUE | expr)
     | expr COLLATE string
     | case_expression
```

Comment:
every time we change this rule, it will like break. We need to rework it to be truly left recursive with precedence corrected.